### ISSUE TITLE
Unable to connect to db using latest 4.4.5 and  neo4j 4.4-community-edition

### ISSUE THREAD

**Issue by hraynaud** [neo4jrb/neo4j-ruby-driver#229]
I have documented the issue in this gitter message and reposting here in case this is the more appropriate location to discuss this issue:
https://matrix.to/#/!ldlqLbTNtjxqvTzpSB:gitter.im/$EzBv4lWwiwAA1e1GVjJpYGoVyaZZfH_IJE2GGChASmY?via=gitter.im&via=matrix.org

Hoping someone can help me figure out what's wrong with my setup.  I have rails app that I was running fine under neo4j-community edition 3.5.  I need to update to community edition 4.4 but I'm unable to get the neo4j-ruby-driver to connect.
My setup is the following:

M1 Mac - Monterey 12.5.1
JDK: 11 (via homebrew)
Ruby: 3..1.2
active\_graph: 11.5.0.beta.3.
neo4j-ruby-driver: 4.4.5

```
dbms.default_database=neo4j
dbms.security.auth_enabled=false


#*****************************************************************
# Network connector configuration
#*****************************************************************
dbms.default_listen_address=0.0.0.0
dbms.default_advertised_address=localhost


# Bolt connector
dbms.connector.bolt.enabled=true
dbms.connector.bolt.tls_level=DISABLED
dbms.connector.bolt.listen_address=:7687
dbms.connector.bolt.advertised_address=:7687


# HTTP Connector. There can be zero or one HTTP connectors.
dbms.connector.http.enabled=true
dbms.connector.http.listen_address=:7474
dbms.connector.http.advertised_address=:7474


# HTTPS Connector. There can be zero or one HTTPS connectors.
dbms.connector.https.enabled=false
#dbms.connector.https.listen_address=:7473
#dbms.connector.https.advertised_address=:7473
```

**neo4j.yml**

```
development:
  url: bolt://localhost:7687
  username: neo4j
  password: password
```

I’m able to start the server and it appears that everything is working fine via the neo4j browser. Which loads fine on the default port :7474

Curl

```
curl localhost:7474
{
  "bolt_routing" : "neo4j://localhost:7687",
  "transaction" : "http://localhost:7474/db/{databaseName}/tx",
  "bolt_direct" : "bolt://localhost:7687",
  "neo4j_version" : "4.4.30",
  "neo4j_edition" : "community"
}%
```

**Netcat**

```
nc -zv localhost 7687
Connection to localhost port 7687 [tcp/*] succeeded!
```

**Cypher Shell:**

```
ensemble git:(main) ✗ ./db/neo4j/development/bin/cypher-shell
Connected to Neo4j using Bolt protocol version 4.4 at neo4j://localhost:7687.
Type :help for a list of available commands or :exit to exit the shell.
Note that Cypher queries must end with a semicolon.
@neo4j> CREATE (a:Item);
0 rows
ready to start consuming query after 115 ms, results consumed after another 0 ms
Added 1 nodes, Added 1 labels
```

Inside a rails console the driver seems to be initialized correctly AFAICT:

```
driver = Neo4j::Driver::GraphDatabase.driver('bolt://localhost:7687',Neo4j::Driver::AuthTokens.basic('neo4j', 'password'))
@session_factory=
#<Neo4j::Driver::Internal::SessionFactoryImpl:0x0000000108a85530
 @connection_provider=
  #<Neo4j::Driver::Internal::DirectConnectionProvider:0x0000000108a856c0
   @address=
    #<Neo4j::Driver::Internal::BoltServerAddress:0x0000000108a8efe0
     @connection_host="localhost",
     @host="localhost",
     @port=7687>,
...
  
       @connector=
        #<Neo4j::Driver::Internal::Async::Connection::ChannelConnectorImpl:0x0000000108a8e090
         @auth_token={:scheme=>"basic", :principal=>"neo4j", :credentials=>"password"},
         @clock=Neo4j::Driver::Internal::Util::Clock::System,
         @connect_timeout_millis=30000,
         
           @context={:address=>"localhost:7687"},
           @server_routing_enabled=false>,
        ...
          @user_agent="neo4j-ruby/4.4.5">,
```

But when I run the same cypher I ran in the cypher shell I get this:

```
driver.session.write_transaction { |tx|'CREATE (a:Item)' }
Neo4j::Driver::Exceptions::ServiceUnavailableException: Connection refused - connect(2) for [::1]:7687
from /Users/herbyraynaud/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/neo4j-ruby-driver-4.4.5/ruby/neo4j/driver/internal/async/connection/channel_connector_impl.rb:34:in `rescue in connect'
Caused by Errno::ECONNREFUSED: Connection refused - connect(2) for [::1]:7687
from /Users/herbyraynaud/.rbenv/versions/3.1.2/lib/ruby/3.1.0/socket.rb:1214:in `__connect_nonblock'
```

**Comment by klobuczek** [comment]
@hraynaud to eliminate any local issues could you please start the neo4j server this way: https://github.com/neo4jrb/activegraph/blob/8e2ba4d117f5702633b0aa7099c71923a100c40d/.github/workflows/test.yml#L35

**Comment by hraynaud** [comment]
@klobuczek thanks for the speedy response. I really appreciate all your effort and contributions and generous offers of assistance. I understand that the issue might be local but I just wanted to confirm a couple of things. Looking at the docker file I notice your pulling a JRuby based image.  

I"m not using JRuby so I'm not sure if that is my problem straight away. Also I'm using a for of the seabolt connector that is compatible with Apple Silicon: [https://github.com/teomores/seabolt-M1](https://github.com/teomores/seabolt-M1)

Is the issue that this setup just won't work?

If that is the case I'm happy to run dockerized but I'm not sure how to set this up exactly. If the issue is with driver on mac shouldn't the problem persist even when connect to the server running on Docker? I will try with docker and let you know what I find.

**Comment by hraynaud** [comment]
@klobuczek thank you again for your generous help. So I was able to get the server running via docker and was able to actually create records in the database. using self committing transactions. here is my progress:
Load neo4j docker:
1. `docker run --name neo4j --env NEO4J_AUTH=neo4j/password --env NEO4J_ACCEPT_LICENSE_AGREEMENT=yes --env NEO4J_dbms_directories_import= -p 7570:7687 -p 7572:7474 -v `pwd`/tmp:/var/lib/neo4j/import --rm neo4j:4.4.30`.  <--- NOTE : changed the exposed port for my test environment. It looks like default internal ports are not able to be change.
2. `rails c -e test`
3.  `driver = Neo4j::Driver::GraphDatabase.driver('bolt://localhost:7687',Neo4j::Driver::AuthTokens.basic('neo4j', 'password'))`
4. `driver.session.run('CREATE (a:Person {name: $name})', name: "herby")`. <-- success :)
5. Running rake neo3j:migrate (and variants) works partially. Some of the migrations start work  and then the system just hangs.   

So it looks like you were right there is something about local environment that is causing the problem. I haven't the slightest clue what it is.  
What's weird is I tested with brand new rails app without any db or migrations and I still could't get the driver to connect. 

That being said I never quite realized that docker was an option and I'm fine using that approach.  Regarding the schema, should I just blow away my schema.yml and all of my migrations and recreate those from scratch? the DB is note huge so this is probably no big deal if that is the path of least resistance.

I'm sure I could figure out the rest of the way forward but if you have any additional advice and/or recommendations to help ensure smoother sailing I would love to hear it. Also, Is it ok to use neo4j 5.x at this point? both 4.x and 5.x worked it seems from docker.

Thank you very much again for your time and dedication to this project. You are awesome.

**Comment by hraynaud** [comment]
@klobuczek You worked your magic once again. I'm not sure exactly what may have happened but it's likely that I may have gotten my conf files screwed up somehow even though I thought I was being extra careful. In any event, my most recent testing was using a fresh rails app with fresh neo4j-community edition install and like I showed in my messages my configs looked legit. 

Well I went back to the old app and suddenly noticed that the configuration hadn't been updated exactly. Both these were commented out again.
```
dbms.default_listen_address=0.0.0.0
dbms.default_advertised_address=localhost
```
What's strange is that in the new app they were not commented out and I was still not able even with these settings uncommented. Well uncommented them and ran the `schema:load` rake task and it just worked.

Honestly that was the only thing I think I had done differently. for some reason I'd forgotten that both schema load and migrate were supported so previously I'd only attempted the migrate.

Thanks again for your help. The only question I have is about running 5.x it looked like it ran fine on docker and I was wondering if it is fully supported for MRI and Silicon M1.  

Thanks again
